catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/ProjectListFragment.java

```diff
@@ -86,6 +90,19 @@
 
 	private static final int REQUEST_IMPORT_PROJECT = 7;
 
+	@Override
+	public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+		View view = super.onCreateView(inflater, container, savedInstanceState);
+		if (getArguments() != null) {
+			Bundle data = getArguments();
+			Intent intent = data.getParcelable("intent");
+			data.putSerializable("IS_IMPORTING_FROM_IMPLICIT_INTENT", true);
+			intent.putExtras(data);
```

Comment:
PLEASE FIX: Since the bundle holds the intent and vise-versa, this could probably lead to a memory loss.
Sure, the Garbage-Collector of Java may realize this, but it's also not a good idea to recycle variables. 
(->Intent->Bundle->)

Reply:
Feel free to refactor function signatures, etc.. to get your code as short and as readable as possible ;)

Reply:
Thanks for pointing these out for me!

Reply:
Thanks for pointing these out for me!